src/main/java/com/jland/website/web/filter/CorsFilter.java

```diff
@@ -0,0 +1,47 @@
+package com.jland.website.web.filter;
+
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class CorsFilter implements Filter {
+
+    @Override
+    public void doFilter(ServletRequest req, ServletResponse res,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletResponse response = (HttpServletResponse) res;
+        HttpServletRequest request = (HttpServletRequest) req;
+
+        if (response != null) {
+            addCorsHeader(response);
+            if ("OPTIONS".equalsIgnoreCase(request.getMethod())) {
+                response.setStatus(HttpServletResponse.SC_OK);
+            } else {
+                filterChain.doFilter(req, res);
+            }
+        }
+    }
+
+    private void addCorsHeader(HttpServletResponse response) {
+        response.addHeader("Access-Control-Allow-Origin", "http://localhost:3000");
```

Comment:
will it work on heroku with localhost?

Reply:
yes, it has been deployed to heroku for test purposes: https://jland-test.herokuapp.com